docs_src/design/adr/system-management/001-docker-image-guidelines.md

```diff
@@ -0,0 +1,34 @@
+# Docker image guidelines
+
+## Status
+
+Pending
+
+## Context
+
+When deploying the EdgeX Docker containers some security measures are recommended to ensure the integrity of the software stack.
+
+## Decision
+
+When deploying Docker images, the following flags should be set for heightened security.
+
+- To avoid escalation of privileges each docker container should use the `--security-opt=no-new-privileges` flag. More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#security-configuration)
+
+- To further prevent privilege escalation attacks the user should be set for the docker container using the `--user=<userid>` or `-u=<userid>` flag. More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#user)
+
+- To avoid a faulty or compromised containers from consuming excess amounts of the host of its resources `resource limits` should be set for each container. more details about `resource limits` can be found [here](https://docs.docker.com/config/containers/resource_constraints/)
+
+- To avoid attackers from writing data to the containers and modifying their files the `--read_only` flag should be set. More details about this flag can be found [here](https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir)
+
+> NOTE: exception
+    If a container is required to have write permission to function, then this flag will not work. For example, the vault requires write permission to write secrets. In this case the `--read_only` flag will not be used.
```

Comment:
Update technical explanation.  Vault needs to run setcap in order to lock pages in memory, but will otherwise run fine if memory page locking is disabled.  In many cases, mounting a writable docker volume in a strategic place will resolve the issue.